src/ocl/convolutionocl.cpp

```diff
@@ -588,12 +639,23 @@ static void ConvForwardCheckNumerics(const Handle& handle,
         return;
     }
 
-    miopen::checkNumericsInput(handle, tensors.xDesc, tensors.x);
-    miopen::checkNumericsInput(handle, tensors.wDesc, tensors.w);
+    bool flag = false;
+
+    flag |= miopen::checkNumericsInput(handle, tensors.xDesc, tensors.x);
```

Comment:
Because either `tensors.x` or `tensors.w` or `tensors.y` could contain `nan`. If any of these three tensor contain `nan` we dump all of them and not just one that encountered `nan`.  Please see [here ](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1752#discussion_r990218840)for more detail.

Reply:
Resolved.

Reply:
But why _bitwise_ (`|`) instead of logical (`||`)?

Reply:
ah! I agree I will replace it with `||`

Reply:
But `|` is actually perfectly portable with booleans. And there is no `||=` form. Any advantage of `||`?

Reply:
If we use `||`
`bool flag = miopen::checkNumericsInput(abc) || miopen::checkNumericsInput(xyz)`  
`miopen::checkNumericsInput(xyz)` will not execute if `miopen::checkNumericsInput(abc)` is true.

solution 1: 
`bool flag1 = miopen::checkNumericsInput(abc);`
`bool flag2 = miopen::checkNumericsInput(xyz);`

`bool flag = flag1 || flag2;`

solution 2:
`bool flag = false;`
`flag |= miopen::checkNumericsInput(abc);`
`flag |= miopen::checkNumericsInput(xyz);`